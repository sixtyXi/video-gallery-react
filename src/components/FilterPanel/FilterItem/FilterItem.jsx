import React, { Fragment } from 'react';

import styles from './FilterItem.scss';

export const FilterItem = ({ name, handleChange, filterName, checked, className }) => {
  return (
    <Fragment>
      <input
        className={styles.filterInput}
        type="radio"
        name={name}
        onChange={handleChange}
        id={filterName}
        value={filterName}
        checked={checked}
      />
      <label className={styles[className]} htmlFor={filterName}>
        {filterName}
      </label>
    </Fragment>
  );
};
