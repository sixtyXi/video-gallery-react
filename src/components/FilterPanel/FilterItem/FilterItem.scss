@import '../../../shared/_colors.scss';
@import '../../../shared/_typography.scss';

.filterInput {
  display: none;
}

label {
  cursor: pointer;
}

.btn {
  margin-left: 10px;
  padding: 0 5px;
  background: $secondary;

  .filterInput:checked + & {
    background: $accent;
  }
}

.label {
  margin-left: 15px;

  .filterInput:checked + & {
    color: $accent;
  }
}
