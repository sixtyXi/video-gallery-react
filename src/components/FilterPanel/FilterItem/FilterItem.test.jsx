import React from 'react';
import { shallow } from 'enzyme';

import { FilterItem } from './FilterItem';

let handleChange;

describe('FilterItem', () => {
  beforeEach(() => {
    handleChange = jest.fn();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders correctly', () => {
    const wrapper = shallow(<FilterItem />);

    expect(wrapper).toMatchSnapshot();
  });

  test('calls *handleChange*', () => {
    const wrapper = shallow(<FilterItem handleChange={handleChange} />);
    const input = wrapper.find('input');

    input.simulate('change');
    expect(handleChange).toHaveBeenCalled();
  });

  test('displays *filterTitle*', () => {
    const filterTitle = 'genre';

    const wrapper = shallow(<FilterItem handleChange={handleChange} filterTitle={filterTitle} />);
    const btnTitle = wrapper.find('label').text();

    expect(btnTitle).toBe(filterTitle);
  });

  test('manages correctly *filterValue*', () => {
    const filterValue = 'release_date';

    const wrapper = shallow(<FilterItem handleChange={handleChange} filterValue={filterValue} />);
    const input = wrapper.find('input');

    const inputId = input.prop('id');
    const inputValue = input.prop('value');

    expect(inputId).toBe(filterValue);
    expect(inputValue).toBe(filterValue);
  });
});
