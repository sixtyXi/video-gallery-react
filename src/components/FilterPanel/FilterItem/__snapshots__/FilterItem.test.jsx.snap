// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterItem renders correctly 1`] = `
<Fragment>
  <input
    checked={false}
    className="filterInput"
    id="vote_average"
    name="filterValue"
    onChange={[MockFunction]}
    type="radio"
    value="vote_average"
  />
  <label
    className="btn"
    htmlFor="vote_average"
  >
    rating
  </label>
</Fragment>
`;
